com.unity.multiplayer.mlapi/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -106,15 +126,31 @@ internal NetworkSerializer __beginSendClientRpc(uint rpcMethodId, ClientRpcParam
         {
             PooledNetworkWriter writer;
 
+            if (clientRpcParams.Send.UpdateStage == NetworkUpdateStage.Unset)
+            {
+                clientRpcParams.Send.UpdateStage = NetworkUpdateLoop.UpdateStage;
+
+                if (clientRpcParams.Send.UpdateStage == NetworkUpdateStage.Initialization)
+                {
+                    clientRpcParams.Send.UpdateStage = NetworkUpdateStage.EarlyUpdate;
+                }
+            }
+
+            if (clientRpcParams.Send.UpdateStage == NetworkUpdateStage.Initialization)
+            {
+                throw new NotSupportedException(
+                    $"{nameof(NetworkUpdateStage.Initialization)} cannot be used as a target for processing RPCs.");
+            }
```

Comment:
Same comment regarding possible use of an assertion.

Reply:
Same comment regarding not exposing this playerloop update stage to the users.

Reply:
Same answer regarding both.

Reply:
@pdeschain the thing is, if you throw an exception and launch the player outside the editor (and, ok, bother to look at logs) the user will see it.  But if you use asserts, they're just elided out and the app happily goes forth as if they weren't there.  I was pro-assert at first until I realized this, BTW

Reply:
Based on what @mattwalsh-unity just said, I'm going to stick to exceptions and mark this and the other conversation both resolved.